Distinguish Between Documentation and Configuration Checks

lkiesow opened the pull request:
This patch stops calling the configuration tests documentation tests and
puts them in a separate workflow with separate rules. This is hopefully
less confusing.

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] be against the correct branch (features can only go into develop)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

lkiesow committed:
Distinguish Between Documentation and Configuration Checks

This patch stops calling the configuration tests documentation tests and
puts them in a separate workflow with separate rules. This is hopefully
less confusing.

LukasKalbertodt commented on docs/checkstyle/check-docs.sh:
```diff
@@ -0,0 +1,30 @@
+  echo "Markdown doc $docs build log:"
+  mkdocs build &> mkdocs.log
+  cat mkdocs.log
+  if grep \
+       -e 'WARNING.*Documentation file' \
+       -e 'pages exist in the docs directory' \
+       -e 'is not found in the documentation files' \
+        mkdocs.log;
```
Why these specific errors only? Why not also fail on "ERROR"? Or just check the exit code of `mkdocs`?

lkiesow replied:
With the modification to this test, it does exactly that contrary to what happened with the `| tee ...` before.
This means, the test now fails if `mkdocs` fails.
These are just additional things to check for.

LukasKalbertodt replied:
Oh yeah, of course, that makes sense!

LukasKalbertodt commented on docs/checkstyle/check-docs.sh:
```diff
@@ -0,0 +1,30 @@
+  cd $docs
+  echo "Markdown doc $docs build log:"
+  mkdocs build &> mkdocs.log
+  cat mkdocs.log
```
Would it make sense to move this line into the `if` so that the log isn't always printed?

LukasKalbertodt commented on docs/checkstyle/check-docs.sh:
```diff
@@ -0,0 +1,30 @@
+for docs in admin developer; do
+  cd $docs
+  echo "Markdown doc $docs build log:"
+  mkdocs build &> mkdocs.log
```
This just silently fails if `mkdocs` or any of the extensions is not installed. Not relevant for CI, but maybe confusing if devs try to execute it locally?

LukasKalbertodt reviewed (commented):
Should `test-documentation-build.yml` be renamed to just `test-documentation.yml`?

I also left a couple inline remarks, but these have nothing to do with your change, but also apply already to the code before this PR. So feel free to ignore them. This PR is certainly an improvement.

lkiesow committed:
More gracefully handle mkdocs errors

This patch hopefully helps debugging mkdocs errors:

- Show mkdocs logs only on errors
- Show an error response if mkdocs itself fails (e.g. the command does
  not exist)

lkiesow commented:
> Should test-documentation-build.yml be renamed to just test-documentation.yml?

Yes, but then we should do that for all the workflows.
So I keep it out of this pull request.

I hopefully addressed all your other comments

LukasKalbertodt commented on docs/checkstyle/check-docs.sh:
```diff
@@ -0,0 +1,34 @@
+      cat mkdocs.log
+      ret=1
+    fi
+  else
+    echo mkdocs exited abnormally.
+    cat mkdocs.log
+  fi
```
```suggestion
  else
    echo mkdocs exited abnormally.
    cat mkdocs.log
    ret=1
  fi
```

Maybe still let the script fail here?

lkiesow replied:
Definitely. Good catch.

LukasKalbertodt reviewed (commented):
Only one thing left!

LukasKalbertodt commented:
By the way: I cannot mark my own comments as "resolved". Weird. Feel free to resolve them yourself.

lkiesow committed:
Update docs/checkstyle/check-docs.sh

Co-authored-by: Lukas Kalbertodt <lukas.kalbertodt@gmail.com>
